src/tox/tox_env/python/api.py

```diff
@@ -108,9 +108,9 @@ def extract_base_python(env_name: str) -> str | None:
         candidates: list[str] = []
         for factor in env_name.split("-"):
             spec = PythonSpec.from_string_spec(factor)
-            if spec.implementation is not None:
-                if spec.implementation.lower() in INTERPRETER_SHORT_NAMES and env_name is not None:
-                    candidates.append(factor)
+            impl = spec.implementation or "python"
```

Comment:
@gaborbernat This change has the unintended (I presume) effect of interpreting a numeric factor (e.g. `2105`) as a candidate base python, breaking previously valid environment names (e.g. `py37-foo-2105`) by raising `ValueError: conflicting factors py37, 2105 in py37-foo-2105`.
Example GitHub workflow build: https://github.com/galaxyproject/planemo/actions/runs/3654111042/jobs/6174245903 for `tox.ini` https://github.com/galaxyproject/planemo/blob/nsoranzo-patch-1/tox.ini

Reply:
That error is a bug 🤣 please fill an issue for it.